### PR TITLE
Update Developer Guide NFR

### PR DESCRIPTION
**Pull request by Chronoxy**

### CODE REVIEW

**Comment by Ma-Xizi** [comment]
LGTM 👍 Good job

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -329,9 +329,31 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+   - Error handling: System should handle up to 90% of incorrect inputs (incorrect date formats, missing fields or etc) without crashing and should provide meaningful error messages to guide users to correct the input
+7. Maintability:
+   - System should have modular components that are easily replaceable or upgradable without affecting the application
+8. Portability:
+   - System must be portable across devices with different operating systems, allowing tutors to install it easily
+9. Quality:
+   - Ease of Use: System should be usable by tutors with minimal computer literacy and include intuitive CLI commands and user-friendly prompts
+   - Testing coverage: Unit tests should cover at least 60% of codebase, ensuring high reliability during future updates
+10. Testability:
+    - Automated Testing: System should support automated unit and integration testing for continuous integration, allowing future updates to be tested without manual intervention
```
[the remark this answers is not in the document]
Comment (Chronoxy) [reply docs/DeveloperGuide.md]: Done 👍

**Comment by NorbertLoh** [comment]
LGTM!

**Review by ethantyh692 — COMMENTED** [review]
LGTM

**Pull request merged**